### PR TITLE
:tada: remove unnecessary unique_id_from_tool in settings.dist.py

### PR DESCRIPTION
**Pull request by manuel-sommer**
This unittest shall help to remove the lines like:
`'AWS Security Hub Scan': ['unique_id_from_tool']` from HASHCODE_FIELDS_PER_SCANNER

The reason is that `['unique_id_from_tool']` should always be DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL

### CODE REVIEW

**Comment by manuel-sommer** [comment]
If this unittest works fine, it should throw an Assertion error for:
```
'AWS Security Hub Scan': ['unique_id_from_tool'],
'Govulncheck Scanner': ['unique_id_from_tool'],
'Azure Security Center Recommendations Scan': ['unique_id_from_tool'],
'Edgescan Scan': ['unique_id_from_tool'],
'Bugcrowd API Import': ['unique_id_from_tool'],
'BlackDuck API': ['unique_id_from_tool'],
'docker-bench-security Scan': ['unique_id_from_tool'],
'Codechecker Report native': ['unique_id_from_tool'],
```

**Comment by manuel-sommer** [comment]
Yes, failed as wrote in comment.

File: unittests/test_settings_dist.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from .dojo_test_case import DojoTestCase, get_unit_tests_path
+import os
+
+basedir = os.path.join(get_unit_tests_path(), '..')
+
+
+class TestSettingsDist(DojoTestCase):
```
Comment (cneill) [inline unittests/test_settings_dist.py:7]: I understand the desire to simplify here - there's no need to specify the hashcode fields if they can be covered by `DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL` - but I am hesitant about adding a test that string-parses the settings file like this. I think replacing the values in `HASHCODE_FIELDS_PER_SCANNER` and `DEDUPLICATION_ALGORITHM_PER_PARSER` as you have done above is sufficient.

Comment (manuel-sommer) [reply unittests/test_settings_dist.py:7]: I understand your thoughts here. I implemented this to help contributors getting feedback on how to use deduplication like this PR shows https://github.com/DefectDojo/django-DefectDojo/pull/9185 

Shall I still remove it, then I can do it.

Comment (cneill) [reply unittests/test_settings_dist.py:7]: I think you can remove this test and just add a brief blurb in the ["how to write a parser" docs](https://github.com/DefectDojo/django-DefectDojo/blob/master/docs/content/en/contributing/how-to-write-a-parser.md#deduplication-algorithm) recommending the use of a pre-defined deduplication algorithm if applicable.

**Review by Maffooch — APPROVED** [review]

**Review by cneill — APPROVED** [review]
Thanks for this! Approved

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Pull request merged**